From f4eb30865f50e175a44ea4077bc7598c60126ce0 Mon Sep 17 00:00:00 2001
From: Tobias <ruck.tobias@gmail.com>
Date: Tue, 13 Apr 2021 09:38:53 -0500
Subject: [PATCH] Fix: Make getblocktemplate create a standard scriptDummy
 (#146)

Otherwise, getblocktemplate fails with "scriptpubkey", as the dummy output created is non-standard.

Rebase provenance: this historical change was adapted onto the pinned
Bitcoin ABC base. Source diff is the recorded Lotus V1 replay commit below.

Historical-source-commit: 3fd45557978593addd7291a221f983805d14a36f
Recorded-replay-commit: b51c3fb5da2c3b2e29d504eff7cbcb19f7b588f7

diff --git a/src/rpc/mining.cpp b/src/rpc/mining.cpp
index f4bf4219f8..0632771d61 100644
--- a/src/rpc/mining.cpp
+++ b/src/rpc/mining.cpp
@@ -1059,8 +1059,9 @@ static RPCHelpMan getblocktemplate() {
                 CBlockIndex *pindexPrevNew = active_chain.Tip();
                 time_start = GetTime();
 
-                // Create new block
-                CScript scriptDummy = CScript() << OP_TRUE;
+                // Create new block. scriptDummy is just for testing block
+                // validity and is not part of the returned template.
+                CScript scriptDummy = CScript() << OP_RETURN;
                 pblocktemplate = BlockAssembler{config, active_chainstate,
                                                 &mempool, node.avalanche.get()}
                                      .CreateNewBlock(scriptDummy);
